shell/app/common/components/tags/index.tsx

```diff
@@ -0,0 +1,162 @@
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import React from 'react';
+import { Tooltip, Popconfirm } from 'antd';
+import { Ellipsis, ErdaIcon } from 'common';
+import i18n from 'i18n';
+import './index.scss';
+
+interface Label {
+  label: string;
+  group?: string;
+  color?: string;
+  checked?: boolean;
+}
+export interface IProps extends Omit<IItemProps, 'label'> {
+  labels: Label[] | Label;
+  showCount?: number;
+  containerClassName?: string;
+}
+
+interface IItemProps {
+  label: Label;
+  maxWidth?: number;
+  colorMap?: Obj<string>;
+  size?: 'small' | 'default';
+  checked?: boolean;
+  deleteConfirm?: boolean;
+  onDelete?: (p: Label) => void;
+}
+
+export const TagItem = (props: IItemProps) => {
+  const { label: _label, size = 'default', maxWidth, onDelete, deleteConfirm = true, colorMap, checked } = props;
+  const { label, color = 'blue' } = _label;
+  const [isChecked, setIsChecked] = React.useState(checked);
+  // TODO: compatible with gray which color is removed now
+  const curColor = color === 'gray' ? 'blue' : color;
+  const style = {
+    maxWidth,
+  };
+
+  const cls = isChecked
+    ? `text-${curColor}-light bg-${curColor}-deep border-0 border-solid border-l-2 border-${curColor}-mid`
+    : `text-${curColor}-deep bg-${curColor}-light border-0 border-solid border-l-2 border-${curColor}-mid`;
+
+  return (
+    <span style={style} className={`tag-default twt-tag-item ${size} ${cls}`}>
+      <div className="flex items-center">
+        <Ellipsis
+          title={label}
+          zIndex={2010} // popconfirm zIndex is bigger than tooltip
+        />
+        {onDelete ? (
+          deleteConfirm ? (
+            <Popconfirm
+              title={`${i18n.t('common:confirm deletion')}?`}
+              arrowPointAtCenter
+              zIndex={2000} //  popconfirm default zIndex=1030, is smaller than tooltip zIndex=1070
+              onConfirm={(e) => {
+                e?.stopPropagation();
+                onDelete(_label);
+              }}
+              onCancel={(e) => e && e.stopPropagation()}
+            >
+              <ErdaIcon size="16" className="cursor-pointer text-default-2 ml-0.5" type="close" color="currentColor" />
+            </Popconfirm>
+          ) : (
+            <ErdaIcon
+              size="16"
+              className="cursor-pointer text-default-2 ml-0.5"
+              type="close"
+              onClick={() => onDelete(_label)}
+            />
+          )
+        ) : null}
+        {checked && (
+          <ErdaIcon
+            size="16"
+            className={`cursor-pointer text-default-2 ml-0.5 text-${color}-light`}
+            type="check"
+            color="currentColor"
+            onClick={() => setIsChecked(!isChecked)}
+          />
+        )}
+      </div>
+    </span>
+  );
+};
+
+const Tags = ({
+  labels: propsLabels,
+  showCount = 2,
```

Comment:
I think `showCount` is very hard to guess the meaning. sounds like a boolean whether should show count number

better be `maxShowCount` if I understand correctly